### ISSUE TITLE
`noarch: python` pins python version to 3.12

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/rattler-build#344]
```yml
# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
context:
  name: polarify
  version: 0.1.3

source:
  url: https://github.com/quantco/polarify/archive/refs/tags/v${{ version }}.tar.gz
  sha256: 93441164c23b764d72c8a66d14b11d5bbd353ed6112ccf3b35efda2a98f9df02

outputs:
  - package:
      name: ${{ name }}
      version: ${{ version }}
    build:
      noarch: python
      script:
        - python -m pip install . --no-deps --ignore-installed -vv --no-build-isolation --disable-pip-version-check
    requirements:
      host:
        - python >=3.9
        - pip
        - hatchling
      run:
        - python >=3.9
        - polars >=0.14.24,<0.20
        # this breaks everything
        # will fail until https://github.com/conda-forge/turbodbc-feedstock/pull/114 is merged
        - turbodbc
    test:
      requires:
        - pip
      commands:
        - pip check
      imports:
        - polarify

about:
  homepage: https://github.com/quantco/polarify
  license: BSD-3-Clause
```

```
Running tests
Creating test environment in "/Users/pavel/polarify/output/bld/rattler-build_polarify_1700738527/work/test"

Resolving for environment specs:
 - pip
 - polarify ==0.1.3 h60d57d3_0


✔ /Users/pavel/polarif [00:00:00] Using cache
✔ /Users/pavel/polarif [00:00:00] Using cache
⠒ conda-forge/osx-arm6 [00:00:00] [────────────────────────────────────────]      0 B @ 0 B/s   
✔ conda-forge/noarch   [00:00:00] Using cache                                                                                                                                                          
✔ conda-forge/osx-arm6 [00:00:00] Using cache
✔ conda-forge/noarch   [00:00:00] Using cache                                                                                                                                                          
Error:   × Failed to setup test environment: Cannot solve the request because of: The following packages are incompatible
  │ |-- polarify ==0.1.3 h60d57d3_0 cannot be installed because there are no viable options:
  │     |-- polarify 0.1.3 would require
  │         |-- python_abi 3.12.* *_cp312, which can be installed with any of the following options:
  │             |-- python_abi 3.12
  │         |-- turbodbc *, which cannot be installed because there are no viable options:
  │             |-- turbodbc 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 |
  │ 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0
  │ | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.9.0 | 4.9.0 | 4.9.0 | 4.9.0 would require
  │                 |-- python >=3.11,<3.12.0a0 *_cpython, which cannot be installed because there are no viable options:
  │                     |-- python 3.11.0 | 3.11.0 | 3.11.0 | 3.11.0 | 3.11.1 | 3.11.2 | 3.11.3 | 3.11.4 | 3.11.5 | 3.11.6 would constrain
  │                         |-- python_abi 3.11.* *_cp311 , which conflicts with any installable versions previously reported
  │             |-- turbodbc 4.2.1 | 4.2.1 | 4.2.1 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.4.0 | 4.4.0 | 4.4.0 | 4.4.0 | 4.5.3 |
  │ 4.5.3 | 4.5.3 | 4.5.3 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 |
  │ 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 |
  │ 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 |
  │ 4.8.0 | 4.8.0 | 4.9.0 | 4.9.0 | 4.9.0 | 4.9.0 would require
  │                 |-- python >=3.10,<3.11.0a0 *_cpython, which cannot be installed because there are no viable options:
  │                     |-- python 3.10.0 | 3.10.0 | 3.10.0 | 3.10.0 | 3.10.0 | 3.10.0 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 |
  │ 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.4 | 3.10.4 | 3.10.5 | 3.10.5 | 3.10.6 | 3.10.6 | 3.10.7 | 3.10.7 | 3.10.8 | 3.10.8 | 3.10.9 | 3.10.10 | 3.10.11 | 3.10.12 | 3.10.13 would constrain
  │                         |-- python_abi 3.10.* *_cp310 , which conflicts with any installable versions previously reported
  │             |-- turbodbc 4.2.0 | 4.2.0 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1
  │ | 4.3.1 | 4.4.0 | 4.4.0 | 4.4.0 | 4.4.0 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5
  │ | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 |
  │ 4.5.10 | 4.5.10 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 |
  │ 4.7.0 | 4.7.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.9.0 | 4.9.0 | 4.9.0 | 4.9.0 would require
  │                 |-- python >=3.9,<3.10.0a0 *_cpython, which cannot be installed because there are no viable options:
  │                     |-- python 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.1rc1 | 3.9.1rc1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.2 | 3.9.4 | 3.9.5 | 3.9.6 | 3.9.6 |
  │ 3.9.7 | 3.9.7 | 3.9.7 | 3.9.7 | 3.9.7 | 3.9.9 | 3.9.9 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.12 | 3.9.12 | 3.9.13 | 3.9.13 | 3.9.14 | 3.9.14 | 3.9.15 | 3.9.15 | 3.9.16 |
  │ 3.9.18 would constrain
  │                         |-- python_abi 3.9.* *_cp39 , which conflicts with any installable versions previously reported
  │             |-- turbodbc 4.2.0 | 4.2.0 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1
  │ | 4.3.1 | 4.4.0 | 4.4.0 | 4.4.0 | 4.4.0 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5
  │ | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 |
  │ 4.5.10 | 4.5.10 would require
  │                 |-- python >=3.8,<3.9.0a0 *_cpython, which cannot be installed because there are no viable options:
  │                     |-- python 3.8.5 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.8 | 3.8.10 | 3.8.10 | 3.8.10 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.13 |
  │ 3.8.13 | 3.8.14 | 3.8.14 | 3.8.15 | 3.8.15 | 3.8.15 | 3.8.16 | 3.8.17 | 3.8.18 would constrain
  │                         |-- python_abi 3.8.* *_cp38 , which conflicts with any installable versions previously reported
  │ 
  ╰─▶ Cannot solve the request because of: The following packages are incompatible
      |-- polarify ==0.1.3 h60d57d3_0 cannot be installed because there are no viable options:
          |-- polarify 0.1.3 would require
              |-- python_abi 3.12.* *_cp312, which can be installed with any of the following options:
                  |-- python_abi 3.12
              |-- turbodbc *, which cannot be installed because there are no viable options:
                  |-- turbodbc 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 |
      4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 |
      4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.9.0 | 4.9.0 | 4.9.0 | 4.9.0 would require
                      |-- python >=3.11,<3.12.0a0 *_cpython, which cannot be installed because there are no viable options:
                          |-- python 3.11.0 | 3.11.0 | 3.11.0 | 3.11.0 | 3.11.1 | 3.11.2 | 3.11.3 | 3.11.4 | 3.11.5 | 3.11.6 would constrain
                              |-- python_abi 3.11.* *_cp311 , which conflicts with any installable versions previously reported
                  |-- turbodbc 4.2.1 | 4.2.1 | 4.2.1 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.4.0 | 4.4.0 | 4.4.0 | 4.4.0 | 4.5.3
      | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6
      | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0
      | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0
      | 4.8.0 | 4.8.0 | 4.9.0 | 4.9.0 | 4.9.0 | 4.9.0 would require
                      |-- python >=3.10,<3.11.0a0 *_cpython, which cannot be installed because there are no viable options:
                          |-- python 3.10.0 | 3.10.0 | 3.10.0 | 3.10.0 | 3.10.0 | 3.10.0 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.1 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2
      | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.2 | 3.10.4 | 3.10.4 | 3.10.5 | 3.10.5 | 3.10.6 | 3.10.6 | 3.10.7 | 3.10.7 | 3.10.8 | 3.10.8 | 3.10.9 | 3.10.10 | 3.10.11 | 3.10.12 | 3.10.13 would constrain
                              |-- python_abi 3.10.* *_cp310 , which conflicts with any installable versions previously reported
                  |-- turbodbc 4.2.0 | 4.2.0 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1
      | 4.3.1 | 4.4.0 | 4.4.0 | 4.4.0 | 4.4.0 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5
      | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 |
      4.5.10 | 4.5.10 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.6.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0 | 4.7.0
      | 4.7.0 | 4.7.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.8.0 | 4.9.0 | 4.9.0 | 4.9.0 | 4.9.0 would require
                      |-- python >=3.9,<3.10.0a0 *_cpython, which cannot be installed because there are no viable options:
                          |-- python 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.1rc1 | 3.9.1rc1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.2 | 3.9.4 | 3.9.5 | 3.9.6 | 3.9.6 |
      3.9.7 | 3.9.7 | 3.9.7 | 3.9.7 | 3.9.7 | 3.9.9 | 3.9.9 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.10 | 3.9.12 | 3.9.12 | 3.9.13 | 3.9.13 | 3.9.14 | 3.9.14 | 3.9.15 | 3.9.15 | 3.9.16 |
      3.9.18 would constrain
                              |-- python_abi 3.9.* *_cp39 , which conflicts with any installable versions previously reported
                  |-- turbodbc 4.2.0 | 4.2.0 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.2.1 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.0 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1 | 4.3.1
      | 4.3.1 | 4.4.0 | 4.4.0 | 4.4.0 | 4.4.0 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.3 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.4 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5
      | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.5 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.6 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.8 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.9 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 | 4.5.10 |
      4.5.10 | 4.5.10 would require
                      |-- python >=3.8,<3.9.0a0 *_cpython, which cannot be installed because there are no viable options:
                          |-- python 3.8.5 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.6 | 3.8.8 | 3.8.10 | 3.8.10 | 3.8.10 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.12 | 3.8.13
      | 3.8.13 | 3.8.14 | 3.8.14 | 3.8.15 | 3.8.15 | 3.8.15 | 3.8.16 | 3.8.17 | 3.8.18 would constrain
                              |-- python_abi 3.8.* *_cp38 , which conflicts with any installable versions previously reported
```

When specifying `python >=3.9,<3.12` instead of `python >=3.9`, everything works.

**Comment by wolfv** [comment]
Looks like this is the culprit: 

```
│ python_abi ┆ 3.12.* *_cp312 (RE of [host: python]) │
```

**Comment by wolfv** [comment]
Hmm ... for some reason the `target_platform` is set to the wrong value. Will debug further!

**Comment by pavelzw** [comment]
Fixed by #347
